### PR TITLE
Dynamically change to bottom navigation

### PR DESCRIPTION
**Pull request by TomMarshBBC**

### CODE REVIEW

File: app/src/main/java/bbc/co/uk/nautilusapp/App.kt [hunk 1]
```diff
@@ -12,6 +13,12 @@ import dagger.android.support.DaggerApplication
  */
 class App : DaggerApplication() {
 
+    companion object {
+        fun coreComponent(context: Context) : CoreComponent {
+            return (context.applicationContext as App).coreComponent
```
Comment (emmaaa) [inline app/src/main/java/bbc/co/uk/nautilusapp/App.kt]: image
?

File: bottom/src/main/java/bbc/co/uk/nautilusapp/bottomnav/ui/BottomNavigationActivity.kt [hunk 2]
```diff
@@ -24,6 +26,8 @@ class BottomNavigationActivity : AppCompatActivity() {
     @Inject
     internal lateinit var navigator: Navigator
 
+    @Inject lateinit var preferences: SharedPreferencesRepository
```
Comment (emmaaa) [inline bottom/src/main/java/bbc/co/uk/nautilusapp/bottomnav/ui/BottomNavigationActivity.kt]: pedantic but the one above is on two lines and autoformat puts it on two lines

File: app/src/main/java/bbc/co/uk/nautilusapp/ui/MainActivity.kt [hunk 3]
```diff
@@ -1,38 +1,49 @@
...
+
     override fun onCreate(savedInstanceState: Bundle?) {
         AndroidInjection.inject(this)
         super.onCreate(savedInstanceState)
 
         remoteConfig.setDefaults(R.xml.default_remote_config_values)
         remoteConfig.fetchAndActivate()
             .addOnCompleteListener {
-                launchFromConfig()
+                launchActivity()
```
Comment (emmaaa) [inline app/src/main/java/bbc/co/uk/nautilusapp/ui/MainActivity.kt]: would be nice to put this chunk into a setConfig() method?
in other activities in the project, (e.g. BottomNavigationActivity) the onCreate is kept quite small (that one has methods abstracted behind setUpToolbar and setUpNavigation methods)

File: core/src/main/java/bbc/co/uk/core/di/CoreComponent.kt [hunk 4]
```diff
@@ -1,13 +1,16 @@
 package bbc.co.uk.core.di
 
 import android.content.Context
+import android.content.SharedPreferences
```
Comment (emmaaa) [inline core/src/main/java/bbc/co/uk/core/di/CoreComponent.kt]: unused import

File: core/src/main/java/bbc/co/uk/core/di/modules/CoreVideoWallModule.kt [hunk 5]
```diff
@@ -2,10 +2,26 @@ package bbc.co.uk.core.di.modules
...
         TrevorVideoWallInteractorModule::class
     ]
 )
-object CoreVideoWallModule
\ No newline at end of file
+object CoreVideoWallModule {
+
+    @Provides
+    @JvmStatic
+    fun providesVideoWallSharedPreferences(
```
Comment (emmaaa) [inline core/src/main/java/bbc/co/uk/core/di/modules/CoreVideoWallModule.kt]: this file is weirdly spread out when things do still fit on one line
seems a lot more readable with 8 and 14 on one line 

```
@Module(includes = [TrevorVideoWallInteractorModule::class])
object CoreVideoWallModule {

    @Provides
    @JvmStatic
    fun providesVideoWallSharedPreferences(preferences: SharedPreferencesRepository): VideoWallSharedPreferences {
        return object : VideoWallSharedPreferences {
            override fun isAutoplayEnabled(): Boolean {
                return preferences.isAutoplayEnabled()
            }
        }
    }

```

File: local-configuration/build.gradle [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+apply plugin: 'kotlin-kapt'
+
+android {
+    compileSdkVersion android_compile_sdk_version
+
+    defaultConfig {
+        minSdkVersion min_sdk_version
+        targetSdkVersion target_sdk_version
+
+        testInstrumentationRunner "androidx.test.runner.AndroidJUnitRunner"
```
Comment (emmaaa) [inline local-configuration/build.gradle]: should use `test_instrumentation_runner`

File: local-configuration/src/main/AndroidManifest.xml [hunk 7]
```diff
@@ -0,0 +1,2 @@
+<manifest xmlns:android="http://schemas.android.com/apk/res/android"
+          package="marsh.tommarsh.local_configuration"/>
```
Comment (emmaaa) [inline local-configuration/src/main/AndroidManifest.xml]: is this right? tom marsh's local configuration?

File: local-configuration/src/main/java/uk/co/bbc/nautilusapp/localconfiguration/SharedPreferencesRepository.kt [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+@Inject constructor(
+    private val context: Context,
+    private val sharedPreferences: SharedPreferences
+) {
+
+    fun isBottomNavEnalbled() = sharedPreferences.getBoolean(context.getString(R.string.key_bottom_nav), false)
+
+    fun isDarkModeEnabled() = sharedPreferences.getBoolean(context.getString(R.string.key_dark_theme), false)
+
+    fun isAutoplayEnabled() =
```
Comment (emmaaa) [inline local-configuration/src/main/java/uk/co/bbc/nautilusapp/localconfiguration/SharedPreferencesRepository.kt]: AutoPlay?

File: local-configuration/src/main/java/uk/co/bbc/nautilusapp/localconfiguration/SharedPreferencesRepository.kt [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+import marsh.tommarsh.local_configuration.R
+import javax.inject.Inject
+
+class SharedPreferencesRepository
+@Inject constructor(
+    private val context: Context,
+    private val sharedPreferences: SharedPreferences
+) {
+
+    fun isBottomNavEnalbled() = sharedPreferences.getBoolean(context.getString(R.string.key_bottom_nav), false)
```
Comment (emmaaa) [inline local-configuration/src/main/java/uk/co/bbc/nautilusapp/localconfiguration/SharedPreferencesRepository.kt]: enalbled

File: bottom/src/main/java/bbc/co/uk/nautilusapp/bottomnav/Navigator.kt [hunk 10]
```diff
@@ -40,7 +40,7 @@ internal class Navigator
     }
 
     private fun openToVideoWall(context: Context, title: String?, id: String) {
-        val intent = VideoWallActivity.newInstance(context, title, id, R.style.VideoWall)
+        val intent = VideoWallActivity.newInstance(context, title, id, bbc.co.uk.core.R.style.VideoWall)
```
[the remark this answers is not in the document]
Comment (TomMarshBBC) [reply bottom/src/main/java/bbc/co/uk/nautilusapp/bottomnav/Navigator.kt]: We're using an R file from another module, so we have to fully qualify one of them :(

File: app/src/main/java/bbc/co/uk/nautilusapp/ui/MainActivity.kt [hunk 11]
```diff
@@ -1,38 +1,49 @@
...
             }
     }
 
-    private fun launchFromConfig() {
-        if (remoteConfig.getBoolean(RIBBON_ENABLED)) {
-            startActivity(Intent(this, RibbonActivity::class.java))
+    private fun launchActivity() {
+        if (preferences.isBottomNavEnalbled()) {
```
[the remark this answers is not in the document]
Comment (TomMarshBBC) [reply app/src/main/java/bbc/co/uk/nautilusapp/ui/MainActivity.kt]: sounds like some sort of adhesive you can get from Screwfix direct.

File: local-configuration/src/main/AndroidManifest.xml (hunk 7 above)
Comment (TomMarshBBC) [reply local-configuration/src/main/AndroidManifest.xml]: got spot 💃

File: app/src/main/java/bbc/co/uk/nautilusapp/ui/MainActivity.kt (hunk 3 above)
Comment (TomMarshBBC) [reply app/src/main/java/bbc/co/uk/nautilusapp/ui/MainActivity.kt]: I don't think it matters too much, but I can do that sure

Comment (TomMarshBBC) [reply app/src/main/java/bbc/co/uk/nautilusapp/ui/MainActivity.kt]: Actually, i'm going to leave this, as i'm going to end up moving it to a remote config module anyway as we have a story for that

**Pull request merged**